webstepper/src/App.tsx

```diff
@@ -3,47 +3,60 @@ import Header from "./Header";
 import { Button, Box, Typography, Stack } from "@mui/material";
 import SvgDisplay from "./SvgDisplay";
 import CodeDisplay from "./CodeDisplay";
+import placeholder from "./placeholder";
 import "./css/styles.css";
 
 export default function App() {
+    if (typeof window === "object" && process.env.NODE_ENV !== "production") {
+        window.svgArray = placeholder.svgArray;
+        window.codeText = placeholder.codeText;
+    }
     const [step, setStep] = useState<number>(0);
-    // TODO: replace this with actual code to display
-    const codeText = `num = 123
-some_string = "Hello, world"
-num2 = 321
-arr = [some_string, "string 123321"]`;
-    const limit = codeText.split("\n").length;
+    const codeText = window.codeText;
+    const limit = Object.keys(window.svgArray).length;
+
     const handleStep = (newStep: number) => {
         setStep(Math.min(Math.max(newStep, 0), limit - 1));
     };
-    const svgPath = `/images/snapshot-${step}.svg`;
 
     return (
         <main className="container">
             <Header />
             <Stack direction="row" spacing={2}>
                 <Box sx={{ width: "40%" }}>
                     <h2>Code</h2>
-                    <Typography>Line: {step + 1}</Typography>
+                    <Typography>
+                        Step {step + 1}/{limit}
+                    </Typography>
                     <Box className="code-display">
                         <CodeDisplay
                             text={codeText}
-                            startingLineNumber={Math.max(step - 10, 1)}
-                            highlightLine={step + 1}
+                            startingLineNumber={Number(
```

Comment:
It doesn't seem like you should need `Number(...)` here, based on the type declaration. Same with `highlightLine` below